pkg/kubelet/kuberuntime/helpers_linux.go

```diff
@@ -77,3 +79,32 @@ func quotaToMilliCPU(quota int64, period int64) int64 {
 	}
 	return (quota * milliCPUToCPU) / period
 }
+
+func subtractOverheadFromResourceConfig(resCfg *cm.ResourceConfig, pod *v1.Pod) *cm.ResourceConfig {
+	if resCfg == nil {
+		return nil
+	}
+
+	rc := *resCfg
+
+	if pod.Spec.Overhead != nil {
+		cpu := pod.Spec.Overhead.Cpu()
+
+		cpuPeriod := int64(*rc.CPUPeriod)
+		totalCPUMilli := sharesToMilliCPU(int64(*rc.CPUShares))
```

Comment:
I don't think `rc.CPUShares` will ever be nil here, but probably safest to check for it anyway.